scripts/migration/lib.sh

```diff
@@ -283,6 +283,48 @@ check_version() {
     fi
   fi
 
+  cluster_version="$(get_apps_version "${1}" >/dev/null 2>&1 || true)"
+  case "${2:-}" in
+  prepare)
+    # Make sure **config** and **cluster** is on the same (old) version
+    if [[ -n "${cluster_version}" ]] && [[ "${cluster_version}" != "${VERSION["${1}-config"]%.*}" ]]; then
+      log_warn "Version mismatch, cluster ${cluster_version}, config ${VERSION["${1}-config"]%.*}"
+      if [[ -t 1 ]]; then
+        log_warn_no_newline "Do you want to continue [y/N]: "
+        read -r reply
+        if [[ "${reply}" != "y" ]]; then
+          exit 1
+        fi
+      else
+        # On mismatch, don't automatically proceed
+        exit 1
+      fi
+    fi
+
+    # Ensure not in the middle of upgrading
+    if get_prepared_version "${1}" &>/dev/null; then
+      log_warn "Migration already in progress"
+      if [[ -t 1 ]]; then
+        log_warn_no_newline "Do you want to continue [y/N]: "
+        read -r reply
+        if [[ "${reply}" != "y" ]]; then
+          exit 1
+        fi
+      else
+        # Don't automatically proceed
+        exit 1
+      fi
+    fi
+    ;;
+    # apply)
+    # In this case, what do we know?
+    # git: new version
+    # config: new version
+    # cluster: old version
+    # cluster: new version
+    # ;;
```

Comment:
Is this left for the future? If so at least write a TODO comment explaining what needs to be done. An issue would be even better.

Reply:
These are mostly notes for myself for something @aarnq wrote in another comment, but I wasn't sure yet what needed to be done.